extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/handler/FSInternetRadioHandler.java

```diff
@@ -0,0 +1,232 @@
+        @Override
+        public void run() {
+            if (radio == null) {
+                // radio is not set, so set all channels to 'undefined'
+                for (Channel channel : getThing().getChannels()) {
+                    updateState(channel.getUID(), UnDefType.UNDEF);
+                }
+                // now let's silently check if it's back online
+                radioLogin(false);
+                return; // if login is successful, this method is called again :-)
+            }
+            try {
+                final boolean radioOn = radio.getPower();
+                for (Channel channel : getThing().getChannels()) {
+                    if (!radioOn && !CHANNEL_POWER.equals(channel.getUID().getId())) {
+                        // if radio is off, set all channels (except for 'POWER') to 'undefined'
+                        updateState(channel.getUID(), UnDefType.UNDEF);
+                    } else if (channel.isLinked()) {
+                        // update all channels that are linked
+                        switch (channel.getUID().getId()) {
+                            case CHANNEL_POWER:
+                                updateState(channel.getUID(), radioOn ? OnOffType.ON : OnOffType.OFF);
+                                break;
+                            case CHANNEL_VOLUME:
+                                updateState(channel.getUID(), DecimalType.valueOf(String.valueOf(radio.getVolume())));
+                                break;
+                            case CHANNEL_MODE:
+                                updateState(channel.getUID(), DecimalType.valueOf(String.valueOf(radio.getMode())));
+                                break;
+                            case CHANNEL_MUTE:
+                                updateState(channel.getUID(), radio.getMuted() ? OnOffType.ON : OnOffType.OFF);
+                                break;
+                            case CHANNEL_PRESET:
+                                // preset is write-only, ignore
+                                break;
+                            case CHANNEL_PLAY_INFO_NAME:
+                                updateState(channel.getUID(), StringType.valueOf(radio.getPlayInfoName()));
+                                break;
+                            case CHANNEL_PLAY_INFO_TEXT:
+                                updateState(channel.getUID(), StringType.valueOf(radio.getPlayInfoText()));
+                                break;
+                            default:
+                                logger.warn("Ignoring unknown channel during update: " + channel.getLabel());
+                        }
+                    }
+                }
+                updateStatus(ThingStatus.ONLINE); // set it back online, maybe it was offline before
+            } catch (Exception e) {
+                String msg = "Communication with radio (update) failed: " + e.getMessage();
+                logger.error(msg);
```

Comment:
remove the logging here, we already get the thing status update (which is logged)